### ISSUE TITLE
Learning outcomes on complex objects

### ISSUE THREAD

**Issue by KulikovNA** [ybkscht/EfficientPose#40]
Hi all, i would like to share the results of training on the geometric body torus object. I believe that this is a complex object for learning due to its symmetry. Without any additional marks, it is quite difficult to determine its orientation. Training was carried out with the following add-ons: hyperparameter phi = 0, epochs = 500, input image size 640x480, pretrained weights efficientdet-0, batch-size = 3 were used, the training set consisted of 8k synthetically generated images. The images were generated using the physically based rendering (PBR) method. The camera position was generated in the range from 20 to 180 centimeters. The material of the object was randomized by roughness, color and reflection parameter. The object itself could be in varying degrees of occlusion. Lighting was also randomized.
Here is the output I got at epoch 500, with the best result coming a little early. Unfortunately I can't watch it: 
`Epoch 500/500
1789/1790 [============================>.] - ETA: 0s - loss: 0.5182 - classificaRunning network: 100% (7920 of 7920) |###| Elapsed Time: 0:08:53 Time:  0:08:53
Parsing annotations: 100% (7920 of 7920) || Elapsed Time: 0:00:00 Time:  0:00:00
100%|██████████████████████████████████████| 7920/7920 [00:32<00:00, 247.40it/s]
7920 instances of class object with average precision: 0.8979
7920 instances of class object with ADD accuracy: 0.0097
7920 instances of class object with ADD-S-Accuracy: 0.5595
7920 instances of class object with 5cm-5degree-Accuracy: 0.0020
class object with Translation Differences in mm: Mean: 30.4905 and Std: 36.1610
class object with Rotation Differences in degree: Mean: 132.0638 and Std: 54.9160
7920 instances of class object with 2d-projection-Accuracy: 0.0302
7920 instances of class object with ADD(-S)-Accuracy: 0.5595
class object with Transformed Point Distances in mm: Mean: 80.4800 and Std: 30.5043
class object with Transformed Symmetric Point Distances in mm: Mean: 17.8650 and Std: 23.0659
class object with Mixed Transformed Point Distances in mm: Mean: 17.8650 and Std: 23.0659
mAP: 0.8979
ADD: 0.0097
ADD-S: 0.5595
5cm_5degree: 0.0020
TranslationErrorMean_in_mm: 30.4905
TranslationErrorStd_in_mm: 36.1610
RotationErrorMean_in_degree: 132.0638
RotationErrorStd_in_degree: 54.9160
2D-Projection: 0.0302
Summed_Translation_Rotation_Error: 253.6313
ADD(-S): 0.5595
AveragePointDistanceMean_in_mm: 80.4800
AveragePointDistanceStd_in_mm: 30.5043
AverageSymmetricPointDistanceMean_in_mm: 17.8650
AverageSymmetricPointDistanceStd_in_mm: 23.0659
MixedAveragePointDistanceMean_in_mm: 17.8650
MixedAveragePointDistanceStd_in_mm: 23.0659

Epoch 00500: ADD-S did not improve from 0.62500
`
Here is an example images:
![000011](https://user-images.githubusercontent.com/70474232/161489240-69633992-382b-48ce-ae6f-d8258d41db62.jpg)

I also conducted training on a geometric body: a cube. The learning outcomes were worse. I would like to know if the only solution for symmetrical objects is adding recognition marks?

**Comment by nikonnext** [comment]
> Hi all, i would like to share the results of training on the geometric body torus object. I believe that this is a complex object for learning due to its symmetry. Without any additional marks, it is quite difficult to determine its orientation. Training was carried out with the following add-ons: hyperparameter phi = 0, epochs = 500, input image size 640x480, pretrained weights efficientdet-0, batch-size = 3 were used, the training set consisted of 8k synthetically generated images. The images were generated using the physically based rendering (PBR) method. The camera position was generated in the range from 20 to 180 centimeters. The material of the object was randomized by roughness, color and reflection parameter. The object itself could be in varying degrees of occlusion. Lighting was also randomized. Here is the output I got at epoch 500, with the best result coming a little early. Unfortunately I can't watch it: `Epoch 500/500 1789/1790 [============================>.] - ETA: 0s - loss: 0.5182 - classificaRunning network: 100% (7920 of 7920) |###| Elapsed Time: 0:08:53 Time: 0:08:53 Parsing annotations: 100% (7920 of 7920) || Elapsed Time: 0:00:00 Time: 0:00:00 100%|██████████████████████████████████████| 7920/7920 [00:32<00:00, 247.40it/s] 7920 instances of class object with average precision: 0.8979 7920 instances of class object with ADD accuracy: 0.0097 7920 instances of class object with ADD-S-Accuracy: 0.5595 7920 instances of class object with 5cm-5degree-Accuracy: 0.0020 class object with Translation Differences in mm: Mean: 30.4905 and Std: 36.1610 class object with Rotation Differences in degree: Mean: 132.0638 and Std: 54.9160 7920 instances of class object with 2d-projection-Accuracy: 0.0302 7920 instances of class object with ADD(-S)-Accuracy: 0.5595 class object with Transformed Point Distances in mm: Mean: 80.4800 and Std: 30.5043 class object with Transformed Symmetric Point Distances in mm: Mean: 17.8650 and Std: 23.0659 class object with Mixed Transformed Point Distances in mm: Mean: 17.8650 and Std: 23.0659 mAP: 0.8979 ADD: 0.0097 ADD-S: 0.5595 5cm_5degree: 0.0020 TranslationErrorMean_in_mm: 30.4905 TranslationErrorStd_in_mm: 36.1610 RotationErrorMean_in_degree: 132.0638 RotationErrorStd_in_degree: 54.9160 2D-Projection: 0.0302 Summed_Translation_Rotation_Error: 253.6313 ADD(-S): 0.5595 AveragePointDistanceMean_in_mm: 80.4800 AveragePointDistanceStd_in_mm: 30.5043 AverageSymmetricPointDistanceMean_in_mm: 17.8650 AverageSymmetricPointDistanceStd_in_mm: 23.0659 MixedAveragePointDistanceMean_in_mm: 17.8650 MixedAveragePointDistanceStd_in_mm: 23.0659
> 
> Epoch 00500: ADD-S did not improve from 0.62500 ` Here is an example images: ![000011](https://user-images.githubusercontent.com/70474232/161489240-69633992-382b-48ce-ae6f-d8258d41db62.jpg)
> 
> I also conducted training on a geometric body: a cube. The learning outcomes were worse. I would like to know if the only solution for symmetrical objects is adding recognition marks?
